Mathlib/Data/Matroid/Constructions.lean

```diff
@@ -0,0 +1,243 @@
+  ⟨hE⟩
+
+@[simp] theorem loopyOn_restrict (E R : Set α) : (loopyOn E) ↾ R = loopyOn R := by
+  refine eq_of_indep_iff_indep_forall rfl ?_
+  simp only [restrict_ground_eq, restrict_indep_iff, loopyOn_indep_iff, and_iff_left_iff_imp]
+  exact fun _ h _ ↦ h
+
+@[simp] theorem empty_base_iff : M.Base ∅ ↔ M = loopyOn M.E := by
+  simp_rw [eq_iff_indep_iff_indep_forall, and_iff_right (show M.E = (loopyOn M.E).E from rfl),
+    loopyOn_indep_iff]
+  refine ⟨fun h I hIE ↦ ⟨fun hI ↦ ?_, ?_⟩, fun h ↦ ?_⟩
+  · rw [h.eq_of_subset_indep hI <| empty_subset _]
+  · rintro rfl
+    exact h.indep.subset <| empty_subset _
+  refine base_iff_maximal_indep.2 ⟨M.empty_indep, fun I hI _ ↦ ?_⟩
+  rwa [eq_comm, ← h I hI.subset_ground]
+
+theorem eq_loopyOn_or_rkPos (M : Matroid α) : M = loopyOn M.E ∨ RkPos M := by
+  rw [← empty_base_iff, rkPos_iff_empty_not_base]; apply em
+
+theorem not_rkPos_iff : ¬RkPos M ↔ M = loopyOn M.E := by
+  rw [rkPos_iff_empty_not_base, not_iff_comm, empty_base_iff]
+
+end LoopyOn
+
+section FreeOn
+
+/-- The `Matroid α` with ground set `E` whose only base is `E`. -/
+def freeOn (E : Set α) : Matroid α := (loopyOn E)﹡
+
+@[simp] theorem freeOn_ground : (freeOn E).E = E := rfl
+
+@[simp] theorem freeOn_dual_eq : (freeOn E)﹡ = loopyOn E := by
+  rw [freeOn, dual_dual]
+
+@[simp] theorem loopyOn_dual_eq : (loopyOn E)﹡ = freeOn E := rfl
+
+@[simp] theorem freeOn_empty (α : Type*) : freeOn (∅ : Set α) = emptyOn α := by
+  simp [freeOn]
+
+@[simp] theorem freeOn_base_iff : (freeOn E).Base B ↔ B = E := by
+  simp only [freeOn, loopyOn_ground, dual_base_iff', loopyOn_base_iff, diff_eq_empty,
+    ← subset_antisymm_iff, eq_comm (a := E)]
+
+@[simp] theorem freeOn_indep_iff : (freeOn E).Indep I ↔ I ⊆ E := by
+  simp [indep_iff_subset_base]
+
+theorem freeOn_indep (hIE : I ⊆ E) : (freeOn E).Indep I :=
+  freeOn_indep_iff.2 hIE
+
+@[simp] theorem freeOn_basis_iff : (freeOn E).Basis I X ↔ I = X ∧ X ⊆ E := by
+  use fun h ↦ ⟨(freeOn_indep h.subset_ground).eq_of_basis h ,h.subset_ground⟩
+  rintro ⟨rfl, hIE⟩
+  exact (freeOn_indep hIE).basis_self
+
+@[simp] theorem freeOn_basis'_iff : (freeOn E).Basis' I X ↔ I = X ∩ E := by
+  rw [basis'_iff_basis_inter_ground, freeOn_basis_iff, freeOn_ground,
+    and_iff_left (inter_subset_right _ _)]
+
+@[simp] theorem eq_freeOn_iff : M = freeOn E ↔ M.E = E ∧ M.Indep E := by
+  refine ⟨?_, fun h ↦ ?_⟩
+  · rintro rfl; simp [Subset.rfl]
+  simp only [eq_iff_indep_iff_indep_forall, freeOn_ground, freeOn_indep_iff, h.1, true_and]
+  exact fun I hIX ↦ iff_of_true (h.2.subset hIX) hIX
+
+theorem ground_indep_iff_eq_freeOn : M.Indep M.E ↔ M = freeOn M.E := by
+  simp
+
+theorem freeOn_restrict (h : R ⊆ E) : (freeOn E) ↾ R = freeOn R := by
+  simp [h, Subset.rfl]
+
+end FreeOn
+
+section TrivialOn
+
+/-- The matroid on `E` whose unique base is the subset `I` of `E`.
+Intended for use when `I ⊆ E`; if this not not the case, then the base is `I ∩ E`. -/
+def trivialOn (I E : Set α) : Matroid α := (freeOn I) ↾ E
```

Comment:
I'm not sure about the name `trivial` here. Usually the trivial matroid is `loopyOn E`, or the uniform matroid of rank 0. I also couldn't come up with a nice name for this matroid, though.

Reply:
What about `uniqueBaseMatroidOn`?

Reply:
or `oneBaseMatroidOn`

Reply:
How about `uniqueBaseOn`? Since other two newly introduced definitions do not include the term `Matroid` in its name, the third name should also follow that rule I guess.